Unit Test code coverage fails

| | |
|:---|:---|
|**_F´ Version_**| 65d960f11a39a4463c6396318602243440cd5846|
|**_Affected Component_**|  All |
---
## Problem Description

Running `fprime-util check --coverage` fails with the following error
```
100% tests passed, 0 tests failed out of 1

Total Test time (real) =   0.01 sec
[100%] Built target Abc_Svc_Time_check
CMakeFiles/Abc_Svc_Time.dir/Time.gcno:cannot open notes file
CMakeFiles/Abc_Svc_Time.dir/Time.gcda:cannot open data file, assuming not executed
CMakeFiles/Abc_Svc_Time.dir/TimeComponentAc.gcno:cannot open notes file
CMakeFiles/Abc_Svc_Time.dir/TimeComponentAc.gcda:cannot open data file, assuming not executed
'/home/ortega/abc/Abc/build-fprime-automatic-native-ut/Abc/Svc/Time/TimeComponentAc.cpp' file is already processed
Error copying file "*.gcov" to "/home/ortega/abc/Abc/Svc/Time/coverage".
make[3]: *** [Abc/Svc/Time/CMakeFiles/Abc_Svc_Time_coverage.dir/build.make:72: Abc/Svc/Time/CMakeFiles/Abc_Svc_Time_coverage] Error 1
make[2]: *** [CMakeFiles/Makefile2:33092: Abc/Svc/Time/CMakeFiles/Abc_Svc_Time_coverage.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:33099: Abc/Svc/Time/CMakeFiles/Abc_Svc_Time_coverage.dir/rule] Error 2
make: *** [Makefile:12491: Abc_Svc_Time_coverage] Error 2
[ERROR] CMake erred with return code 2

```
## How to Reproduce

1. cd into a component's directory
2. Run `fprime-util check --coverage`

## Expected Behavior

I expected gcov files in my component's directory.

--- a/cmake/target/build.cmake
+++ b/cmake/target/build.cmake
@@ -91,6 +91,11 @@ function(build_setup_build_module MODULE SOURCES GENERATED EXCLUDED_SOURCES DEPE
         endif()
     endforeach()
     set_property(TARGET "${MODULE}" PROPERTY FPRIME_TARGET_DEPENDENCIES ${TARGET_DEPENDENCIES})
+    # Special flags applied to modules when compiling with testing enabled
+    if (BUILD_TESTING)
+        target_compile_options("${MODULE}" PRIVATE ${FPRIME_TESTING_REQUIRED_COMPILE_FLAGS})
+        target_link_libraries("${MODULE}" PRIVATE ${FPRIME_TESTING_REQUIRED_LINK_FLAGS})
+    endif()
 endfunction()
 
 ####
@@ -120,11 +125,6 @@ function(build_add_module_target MODULE TARGET SOURCES DEPENDENCIES)
     run_ac_set("${SOURCES}" ${CUSTOM_AUTOCODERS})
     resolve_dependencies(RESOLVED ${DEPENDENCIES} ${AC_DEPENDENCIES} )
     build_setup_build_module("${MODULE}" "${SOURCES}" "${AC_GENERATED}" "${AC_SOURCES}" "${RESOLVED}")
-    # Special flags applied to modules when compiling with testing enabled
-    if (BUILD_TESTING)
-        target_compile_options("${MODULE}" PRIVATE ${FPRIME_TESTING_REQUIRED_COMPILE_FLAGS})
-        target_link_libraries("${MODULE}" PRIVATE ${FPRIME_TESTING_REQUIRED_LINK_FLAGS})
-    endif()
 
     if (CMAKE_DEBUG_OUTPUT)
         introspect("${MODULE}")

--- a/cmake/target/check.cmake
+++ b/cmake/target/check.cmake
@@ -11,9 +11,7 @@
 # - **TARGET_NAME:** target name to be generated
 ####
 function(check_add_global_target TARGET_NAME)
-    add_custom_target(${TARGET_NAME}
-            COMMAND ${CMAKE_COMMAND} -E chdir ${CMAKE_BINARY_DIR} find . -name "*.gcda" -delete
-            COMMAND ${CMAKE_CTEST_COMMAND})
+    add_custom_target(${TARGET_NAME} COMMAND ${CMAKE_CTEST_COMMAND})
 endfunction(check_add_global_target)
 
 ####
@@ -33,12 +31,18 @@ function(check_add_deployment_target MODULE TARGET SOURCES DEPENDENCIES FULL_DEP
         get_property(DEPENDENCY_UTS TARGET "${DEPENDENCY}" PROPERTY FPRIME_UTS)
         list(APPEND ALL_UTS ${DEPENDENCY_UTS})
     endforeach()
-    string(REPLACE ";" "\\|" JOINED_UTS "${ALL_UTS}")
-    add_custom_target(${MODULE}_${TARGET_NAME}
-        COMMAND ${CMAKE_COMMAND} -E chdir ${CMAKE_BINARY_DIR} find . -name "*.gcda" -delete
-        COMMAND ${CMAKE_CTEST_COMMAND} -R "${JOINED_UTS}"
-        DEPENDS ${ALL_UTS}
-    )
+    # Only run deployment UTs when some are defined
+    if (ALL_UTS)
+        string(REPLACE ";" "\\|" JOINED_UTS "${ALL_UTS}")
+        add_custom_target(${MODULE}_${TARGET_NAME}
+            COMMAND ${CMAKE_CTEST_COMMAND} -R "${JOINED_UTS}"
+            DEPENDS ${ALL_UTS}
+        )
+    else()
+        add_custom_target(${MODULE}_${TARGET_NAME}
+            COMMAND ${CMAKE_COMMAND} -E echo "No unit tests defined for ${MODULE}"
+        )
+    endif()
 endfunction()
 
 ####
@@ -58,7 +62,6 @@ function(check_add_module_target MODULE_NAME TARGET_NAME SOURCE_FILES DEPENDENCI
     elseif (NOT TARGET ${MODULE_NAME}_${TARGET_NAME})
         add_custom_target(
             "${MODULE_NAME}_${TARGET_NAME}"
-            COMMAND ${CMAKE_COMMAND} -E chdir ${CMAKE_BINARY_DIR} find . -name "*.gcda" -delete
             COMMAND ${CMAKE_CTEST_COMMAND} --verbose
         )
     endif()

--- a/requirements.txt
+++ b/requirements.txt
@@ -18,7 +18,7 @@ Flask-RESTful==0.3.9
 fprime-fpp==1.2.0
 fprime-gds==3.2.0
 fprime-tools==3.2.0
-gcovr==5.2
+gcovr==6.0
 idna==3.4
 importlib-metadata==4.13.0
 iniconfig==2.0.0

